src/lib/locationHelpers.ts

```diff
@@ -15,7 +15,7 @@ export const getLocationArgs = async (
 ) => {
   let location = near
 
-  if (ip) {
+  if (!location && ip) {
```

Comment:
If `near` is passed and `location` already set,   we don't want to do an IP geo lookup.